packages/autodetect/api-report.md

```diff
@@ -0,0 +1,19 @@
+## API Report File for "@backstage/autodetect"
```

Comment:
Naming is quite vague and currently feels a bit like a misc package. Do we imagine this to be bigger than detecting plugins? Otherwise lets try and add more specificity to what the package is intended to provide.